Setup Java Logging Framework #17

n-machiavelli opened the pull request:
Add logback files for dev and prod.
logback uses sl4j library.

n-machiavelli committed:
Setup Java Logging Framework #17

Add logback files for dev and prod.
logback uses sl4j library.

n-machiavelli committed:
Rest API - User - Add Title #112

Code4SocialGood commented:
1. Should we not check in `logFile.2017-04-07.log` to the repository? - DELETED
2. Where will be the log file dumped? Could we create a directory just for log files? - the logback file will create logs directory now.

felipealvesgnu commented on src/main/java/org/c4sg/entity/User.java:
```diff
@@ -90,6 +90,9 @@
 	private Date updatedTime;
+	
+    @Column(name = "title")
+    private String title;	
```
Pay attention on indentation

n-machiavelli replied:
Oh I must have used spaces instead of a tab. Thanks.

felipealvesgnu replied:
solve it and push the code again

n-machiavelli committed:
Indent User.java.
Remove logfile from repo. Update logback-prod to create logs folder

felipealvesgnu commented:
@n-machiavelli Can you answer @Code4SocialGood 's questions ?

n-machiavelli commented:
Should we not check in logFile.2017-04-07.log to the repository? - DELETED as per Slack convo
Where will be the log file dumped? Could we create a directory just for log files? - the logback file will create logs directory now as per slack convo

Code4SocialGood commented:
How to get a log file generated? I called getActiveUser API fromo http://localhost:8080/swagger-ui.html#!/user/getActiveUsersUsingGET, which calls  LOGGER.debug("**************All**************"); But I didn't get a log file generated locally.

![image](https://cloud.githubusercontent.com/assets/24813620/24920960/cd3c7502-1eae-11e7-92f1-6475eaf13a0f.png)

n-machiavelli committed:
Automate logback file selection using appropriate application.properties

n-machiavelli committed:
Automate logback file selection using appropriate application.properties for prod

n-machiavelli commented:
This has been resolved. Note that you won't have "log.debug" in the log file as prod is set to INFO level. but if you have log.info for testing you will find it:
![image](https://cloud.githubusercontent.com/assets/5101665/25006124/472df304-2021-11e7-917c-44d5bb4c8209.png)
![image](https://cloud.githubusercontent.com/assets/5101665/25006154/65d41716-2021-11e7-9666-41efccdf7b11.png)

Code4SocialGood commented:
1. Are we using log4j2 or logback? The naming of 3 logging config files is confusing: log4j2.xml, logback-dev.xml, logback-prod.xml
2. Local Development Environment: After starting Spring Boot, a log file is generated: \logs\logFile.2017-04-13.log. Please change the config so that all the logging are sent to console. No log file for local dev.
3. Test server environment and Prod environment: Log level is DEBUG for test server and INFO for prod. Logging is sent to file appender. What do I need to change so that test/prod server config will be used?

n-machiavelli committed:
Delete log4j2.xml as we are using logback
Enable logfile for only dev

n-machiavelli commented:
1. Are we using log4j2 or logback? We are using logback. The log4j2 file was there before and I have just removed it and retested.
2. I have fixed this in application-dev.properties. No log file for local dev.
3. This is setup by setting the appropriate logback file in the last line of application-dev.properties and application-prod.properties

![image](https://cloud.githubusercontent.com/assets/5101665/25034447/3560e68c-20ac-11e7-918f-e5b3622c1791.png)

![image](https://cloud.githubusercontent.com/assets/5101665/25034453/4460f41a-20ac-11e7-8c7d-20e941880d39.png)
